diff -Bubr ../busybox-1.00+/archival/libunarchive/decompress_bunzip2.c ./archival/libunarchive/decompress_bunzip2.c
--- ../busybox-1.00+/archival/libunarchive/decompress_bunzip2.c	2004-08-27 17:43:05.000000000 -0700
+++ ./archival/libunarchive/decompress_bunzip2.c	2009-06-07 15:10:37.000000000 -0700
@@ -136,7 +136,7 @@
 {
 	/* Note: Ignore the warning about hufGroup, base and limit being used uninitialized.
 	 * They will be initialized on the fist pass of the loop. */
-	struct group_data *hufGroup;
+	struct group_data *hufGroup=NULL;
 	int dbufCount,nextSym,dbufSize,groupCount,*base,*limit,selector,
 		i,j,k,t,runPos,symCount,symTotal,nSelectors,byteCount[256];
 	unsigned char uc, symToByte[256], mtfSymbol[256], *selectors;
Only in ./archival/libunarchive: decompress_bunzip2.c.orig
diff -Bubr ../busybox-1.00+/archival/tar.c ./archival/tar.c
--- ../busybox-1.00+/archival/tar.c	2004-08-26 15:18:56.000000000 -0700
+++ ./archival/tar.c	2009-06-07 15:19:14.000000000 -0700
@@ -48,8 +48,10 @@
 #include <errno.h>
 #include <signal.h>
 #include <sys/wait.h>
+#ifndef NESTEDVM
 #include <sys/socket.h>
 #include <sys/sysmacros.h>     /* major() and minor() */
+#endif
 #include "unarchive.h"
 #include "busybox.h"
 
Only in ./archival: tar.c.orig
Only in .: busybox-1.00-pre9
Only in .: .config
Only in .: .config.cmd
Only in .: .config.old
diff -Bubr ../busybox-1.00+/coreutils/cal.c ./coreutils/cal.c
--- ../busybox-1.00+/coreutils/cal.c	2004-04-14 10:51:09.000000000 -0700
+++ ./coreutils/cal.c	2009-06-07 15:08:50.000000000 -0700
@@ -31,7 +31,9 @@
 
 #include <sys/types.h>
 #include <ctype.h>
+#ifndef NESTEDVM
 #include <err.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -Bubr ../busybox-1.00+/coreutils/date.c ./coreutils/date.c
--- ../busybox-1.00+/coreutils/date.c	2004-10-11 13:52:16.000000000 -0700
+++ ./coreutils/date.c	2009-06-07 15:08:50.000000000 -0700
@@ -236,7 +236,11 @@
 		}
 
 		/* if setting time, set it */
+#ifndef NESTEDVM
 		if (set_time && (stime(&tm) < 0)) {
+#else
+        if(0) {
+#endif
 			bb_perror_msg("cannot set date");
 		}
 	}
Only in ./coreutils: date.c.orig
diff -Bubr ../busybox-1.00+/coreutils/df.c ./coreutils/df.c
--- ../busybox-1.00+/coreutils/df.c	2004-08-02 17:14:01.000000000 -0700
+++ ./coreutils/df.c	2009-06-07 15:20:57.000000000 -0700
@@ -35,8 +35,10 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#ifndef NESTEDVM
 #include <mntent.h>
 #include <sys/vfs.h>
+#endif
 #include "busybox.h"
 
 #ifndef CONFIG_FEATURE_HUMAN_READABLE
@@ -46,6 +48,7 @@
 }
 #endif
 
+#ifndef NESTEDVM
 extern int df_main(int argc, char **argv)
 {
 	long blocks_used;
@@ -160,6 +163,7 @@
 
 	bb_fflush_stdout_and_exit(status);
 }
+#endif
 
 /*
 Local Variables:
diff -Bubr ../busybox-1.00+/coreutils/dos2unix.c ./coreutils/dos2unix.c
--- ../busybox-1.00+/coreutils/dos2unix.c	2004-05-05 12:39:21.000000000 -0700
+++ ./coreutils/dos2unix.c	2009-06-07 15:08:50.000000000 -0700
@@ -30,7 +30,9 @@
 #include <string.h>
 #include <getopt.h>
 #include <unistd.h>
+#ifndef NESTEDVM
 #include <stdint.h>
+#endif
 #include <fcntl.h>
 #include <sys/time.h>
 #include "busybox.h"
diff -Bubr ../busybox-1.00+/coreutils/env.c ./coreutils/env.c
--- ../busybox-1.00+/coreutils/env.c	2004-04-16 08:02:10.000000000 -0700
+++ ./coreutils/env.c	2009-06-07 15:08:50.000000000 -0700
@@ -50,12 +50,13 @@
 #include <getopt.h>
 #include "busybox.h"
 
-
+#ifdef NESTEDVM
 static const struct option env_long_options[] = {
 	{ "ignore-environment", 0, NULL, 'i' },
 	{ "unset", 1, NULL, 'u' },
 	{ 0, 0, 0, 0 }
 };
+#endif
 
 extern int env_main(int argc, char** argv)
 {
diff -Bubr ../busybox-1.00+/coreutils/expr.c ./coreutils/expr.c
--- ../busybox-1.00+/coreutils/expr.c	2004-04-14 10:51:09.000000000 -0700
+++ ./coreutils/expr.c	2009-06-07 15:22:16.000000000 -0700
@@ -244,6 +244,7 @@
 
 static VALUE *docolon (VALUE *sv, VALUE *pv)
 {
+#ifndef NESTEDVM
 	VALUE *v;
 	const char *errmsg;
 	struct re_pattern_buffer re_buffer;
@@ -291,6 +292,9 @@
 	}
 	free (re_buffer.buffer);
 	return v;
+#else
+        return NULL;
+#endif
 }
 
 /* Handle bare operands and ( expr ) syntax.  */
diff -Bubr ../busybox-1.00+/coreutils/ls.c ./coreutils/ls.c
--- ../busybox-1.00+/coreutils/ls.c	2004-09-23 19:04:13.000000000 -0700
+++ ./coreutils/ls.c	2009-06-07 15:22:37.000000000 -0700
@@ -59,9 +59,11 @@
 #include <stdlib.h>
 #include <fcntl.h>
 #include <signal.h>
+#ifndef NESTEDVM
 #include <termios.h>
 #include <sys/ioctl.h>
 #include <sys/sysmacros.h>     /* major() and minor() */
+#endif
 #include "busybox.h"
 #ifdef CONFIG_SELINUX
 #include <fs_secure.h>
@@ -984,8 +986,11 @@
 #endif
 
 #ifdef CONFIG_FEATURE_LS_COLOR
-	if (isatty(STDOUT_FILENO))
+    {
+        char *term = getenv("TERM");
+        if (isatty(STDOUT_FILENO) && term && *term)
 		show_color = 1;
+    }
 #endif
 
 	/* process options */
Only in ./coreutils: ls.c.orig
diff -Bubr ../busybox-1.00+/coreutils/md5_sha1_sum.c ./coreutils/md5_sha1_sum.c
--- ../busybox-1.00+/coreutils/md5_sha1_sum.c	2004-04-14 10:51:09.000000000 -0700
+++ ./coreutils/md5_sha1_sum.c	2009-06-07 15:08:50.000000000 -0700
@@ -20,7 +20,9 @@
 #include <fcntl.h>
 #include <limits.h>
 #include <stdio.h>
+#ifndef NESTEDVM
 #include <stdint.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
diff -Bubr ../busybox-1.00+/coreutils/rmdir.c ./coreutils/rmdir.c
--- ../busybox-1.00+/coreutils/rmdir.c	2004-03-15 00:28:21.000000000 -0800
+++ ./coreutils/rmdir.c	2009-06-07 15:08:50.000000000 -0700
@@ -25,7 +25,9 @@
 
 #include <stdlib.h>
 #include <unistd.h>
+#ifndef NESTEDVM
 #include <libgen.h>
+#endif
 #include "busybox.h"
 
 extern int rmdir_main(int argc, char **argv)
diff -Bubr ../busybox-1.00+/editors/vi.c ./editors/vi.c
--- ../busybox-1.00+/editors/vi.c	2004-08-19 12:15:06.000000000 -0700
+++ ./editors/vi.c	2009-06-07 15:24:23.000000000 -0700
@@ -68,7 +68,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifndef NESTEDVM
 #include <termios.h>
+#endif
 #include <unistd.h>
 #include <sys/ioctl.h>
 #include <sys/time.h>
@@ -183,7 +185,9 @@
 static Byte *screenbegin;	// index into text[], of top line on the screen
 static Byte *dot;		// where all the action takes place
 static int tabstop;
+#ifndef NESTEDVM
 static struct termios term_orig, term_vi;	// remember what the cooked mode was
+#endif
 static Byte erase_char;         // the users erase character
 static Byte last_input_char;    // last char read from user
 static Byte last_forward_char;  // last char searched for with 'f'
@@ -2113,6 +2117,7 @@
 //----- Set terminal attributes --------------------------------
 static void rawmode(void)
 {
+#ifndef NESTEDVM
 	tcgetattr(0, &term_orig);
 	term_vi = term_orig;
 	term_vi.c_lflag &= (~ICANON & ~ECHO);	// leave ISIG ON- allow intr's
@@ -2122,12 +2127,15 @@
 	term_vi.c_cc[VTIME] = 0;
 	erase_char = term_vi.c_cc[VERASE];
 	tcsetattr(0, TCSANOW, &term_vi);
+#endif
 }
 
 static void cookmode(void)
 {
 	fflush(stdout);
+#ifndef NESTEDVM
 	tcsetattr(0, TCSANOW, &term_orig);
+#endif
 }
 
 //----- Come here when we get a window resize signal ---------
diff -Bubr ../busybox-1.00+/include/busybox.h ./include/busybox.h
--- ../busybox-1.00+/include/busybox.h	2004-03-15 00:28:38.000000000 -0800
+++ ./include/busybox.h	2009-06-07 15:08:50.000000000 -0700
@@ -33,7 +33,7 @@
 #include <sys/types.h>
 
 #if __GNU_LIBRARY__ < 5
-#ifndef __dietlibc__
+#if !defined(__dietlibc__) && !defined(NESTEDVM)
 #error "Sorry, libc5 is not supported"
 #endif
 #endif
@@ -48,7 +48,9 @@
 #include <dmalloc.h>
 #endif
 
+#ifndef NESTEDVM
 #include <features.h>
+#endif
 
 /* Pull in the utility routines from libbb */
 #include "libbb.h"
diff -Bubr ../busybox-1.00+/include/grp_.h ./include/grp_.h
--- ../busybox-1.00+/include/grp_.h	2004-07-15 05:53:49.000000000 -0700
+++ ./include/grp_.h	2009-06-07 15:08:50.000000000 -0700
@@ -31,7 +31,9 @@
 
 
 #include <sys/types.h>
+#ifndef NESTEDVM
 #include <features.h>
+#endif
 #include <stdio.h>
 
 
Only in ./include: grp_.h.orig
diff -Bubr ../busybox-1.00+/include/libbb.h ./include/libbb.h
--- ../busybox-1.00+/include/libbb.h	2004-09-14 20:04:07.000000000 -0700
+++ ./include/libbb.h	2009-06-07 15:08:50.000000000 -0700
@@ -24,22 +24,68 @@
 #ifndef	__LIBCONFIG_H__
 #define	__LIBCONFIG_H__    1
 
+
+#ifdef NESTEDVM
+#include <sys/cdefs.h>
+#include <sys/fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#define WCOREDUMP(x) 0
+
+typedef char int8_t;
+typedef unsigned char uint8_t;
+typedef short int16_t;
+typedef unsigned short uint16_t;
+typedef int int32_t;
+typedef unsigned int uint32_t;
+typedef long long int64_t;
+typedef unsigned long long uint64_t;
+
+struct option {
+	const char *name;
+	int has_arg;
+	int *flag;
+	int val;
+};
+
+#include <stdarg.h>
+
+static char *strchrnul(const char *s, int c) {
+    char *ret;
+    ret = strchr(s,c);
+    return ret ? ret : (char*)s + strlen(s);
+}
+
+static char *stpcpy(char *dest, const char *src) {
+    while((*dest = *src) != '\0') { dest++; src++; }
+    return dest;
+}
+
+#endif
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdarg.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <regex.h>
+#ifndef NESTEDVM
 #include <termios.h>
 #include <stdint.h>
+#endif
 
 #include <netdb.h>
+#include <netinet/in.h>
 
 #ifdef DMALLOC
 #include <dmalloc.h>
 #endif
 
+#ifndef NESTEDVM
 #include <features.h>
+#endif
 
 #include "config.h"
 #ifdef CONFIG_SELINUX
@@ -123,8 +169,10 @@
 extern int bb_parse_mode( const char* s, mode_t* theMode);
 extern long bb_xgetlarg(const char *arg, int base, long lower, long upper);
 
+#ifndef NESTEDVM
 extern unsigned long bb_baud_to_value(speed_t speed);
 extern speed_t bb_value_to_baud(unsigned long value);
+#endif
 
 extern int get_kernel_revision(void);
 
diff -Bubr ../busybox-1.00+/include/pwd_.h ./include/pwd_.h
--- ../busybox-1.00+/include/pwd_.h	2004-07-15 05:53:49.000000000 -0700
+++ ./include/pwd_.h	2009-06-07 15:08:50.000000000 -0700
@@ -29,7 +29,9 @@
 #define	_PWD_H	1
 
 #include <sys/types.h>
+#ifndef NESTEDVM
 #include <features.h>
+#endif
 #include <stdio.h>
 
 /* The passwd structure.  */
Only in ./include: pwd_.h.orig
diff -Bubr ../busybox-1.00+/init/halt.c ./init/halt.c
--- ../busybox-1.00+/init/halt.c	2004-03-15 00:28:40.000000000 -0800
+++ ./init/halt.c	2009-06-07 15:24:41.000000000 -0700
@@ -24,7 +24,9 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <getopt.h>
+#ifndef NESTEDVM
 #include <sys/reboot.h>
+#endif
 #include "busybox.h"
 #include "init_shared.h"
 
Only in ./init: .#init.c
diff -Bubr ../busybox-1.00+/init/init.c ./init/init.c
--- ../busybox-1.00+/init/init.c	2004-10-08 01:21:54.000000000 -0700
+++ ./init/init.c	2009-06-07 15:26:03.000000000 -0700
@@ -34,23 +34,31 @@
 #include <signal.h>
 #include <stdarg.h>
 #include <string.h>
+#ifndef NESTEDVM
 #include <termios.h>
+#endif
 #include <unistd.h>
 #include <limits.h>
 #include <sys/fcntl.h>
 #include <sys/ioctl.h>
+#ifndef NESTEDVM
 #include <sys/mount.h>
+#endif
 #include <sys/types.h>
 #include <sys/wait.h>
+#ifndef NESTEDVM
 #include <sys/reboot.h>
+#endif
 #include "busybox.h"
 
 #include "init_shared.h"
 
 
+#ifndef NESTEDVM
 #ifdef CONFIG_SYSLOGD
 # include <sys/syslog.h>
 #endif
+#endif
 
 
 #define INIT_BUFFS_SIZE 256
@@ -675,7 +683,9 @@
 	 * linux/kernel/sys.c, which can cause the machine to panic when
 	 * the init process is killed.... */
 	if ((pid = fork()) == 0) {
+#ifndef NESTEDVM
 		reboot(magic);
+#endif
 		_exit(0);
 	}
 	waitpid (pid, NULL, 0);
diff -Bubr ../busybox-1.00+/libbb/getopt_ulflags.c ./libbb/getopt_ulflags.c
--- ../busybox-1.00+/libbb/getopt_ulflags.c	2004-02-05 05:49:29.000000000 -0800
+++ ./libbb/getopt_ulflags.c	2009-06-07 15:08:50.000000000 -0700
@@ -150,8 +150,14 @@
     s--;
   }
 
-  while ((c = getopt_long (argc, argv, applet_opts,
-			    bb_applet_long_options, NULL)) > 0) {
+    
+  while ((c = 
+#ifdef NESTEDVM
+          getopt(argc,argv,applet_opts)
+#else
+          getopt_long (argc, argv, applet_opts,bb_applet_long_options, NULL) 
+#endif
+  ) > 0) {
 	for (on_off = complementaly; on_off->opt != c; on_off++) {
 	    if(!on_off->opt)
 			bb_show_usage ();
diff -Bubr ../busybox-1.00+/libbb/get_terminal_width_height.c ./libbb/get_terminal_width_height.c
--- ../busybox-1.00+/libbb/get_terminal_width_height.c	2004-03-23 15:15:35.000000000 -0800
+++ ./libbb/get_terminal_width_height.c	2009-06-07 15:08:50.000000000 -0700
@@ -24,8 +24,10 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <unistd.h>
+#ifndef NESTEDVM
 #include <termios.h>
 #include <sys/ioctl.h>
+#endif
 #include "busybox.h"
 
 /* It is perfectly ok to pass in a NULL for either width or for
@@ -34,7 +36,11 @@
  * which case you will always get 80x24 */
 void get_terminal_width_height(int fd, int *width, int *height)
 {
+#ifdef NESTEDVM
+    struct { int ws_row; int ws_col; } win = { 0, 0};
+#else
 	struct winsize win = { 0, 0, 0, 0 };
+#endif
 #ifdef CONFIG_FEATURE_AUTOWIDTH
 	if (ioctl(fd, TIOCGWINSZ, &win) != 0) {
 		win.ws_row = 24;
diff -Bubr ../busybox-1.00+/libbb/hash_fd.c ./libbb/hash_fd.c
--- ../busybox-1.00+/libbb/hash_fd.c	2004-03-15 00:28:42.000000000 -0800
+++ ./libbb/hash_fd.c	2009-06-07 15:08:50.000000000 -0700
@@ -20,18 +20,27 @@
  *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
+#ifndef NESTEDVM
 #include <byteswap.h>
 #include <endian.h>
+#endif
 #include <fcntl.h>
 #include <limits.h>
 #include <stdio.h>
+#ifndef NESTEDVM
 #include <stdint.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 
 #include "busybox.h"
 
+#ifdef NESTEDVM
+#define __BYTE_ORDER BYTE_ORDER
+#define __LITTLE_ENDIAN LITTLE_ENDIAN
+#define __BIG_ENDIAN BIG_ENDIAN
+#endif
 
 #ifdef CONFIG_SHA1SUM
 /*
Only in ./libbb: Makefile.in.orig
diff -Bubr ../busybox-1.00+/libbb/procps.c ./libbb/procps.c
--- ../busybox-1.00+/libbb/procps.c	2004-08-26 15:18:58.000000000 -0700
+++ ./libbb/procps.c	2009-06-07 15:08:50.000000000 -0700
@@ -12,7 +12,9 @@
 #include <string.h>
 #include <stdlib.h>
 #include <unistd.h>
+#ifndef NESTEDVM
 #include <asm/page.h>
+#endif
 
 #include "libbb.h"
 
diff -Bubr ../busybox-1.00+/libbb/run_shell.c ./libbb/run_shell.c
--- ../busybox-1.00+/libbb/run_shell.c	2004-03-15 00:28:43.000000000 -0800
+++ ./libbb/run_shell.c	2009-06-07 15:08:50.000000000 -0700
@@ -33,7 +33,9 @@
 #include <unistd.h>
 #include <string.h>
 #include <stdlib.h>
+#ifndef NESTEDVM
 #include <syslog.h>
+#endif
 #include <ctype.h>
 #include "libbb.h"
 #ifdef CONFIG_SELINUX
diff -Bubr ../busybox-1.00+/libbb/u_signal_names.c ./libbb/u_signal_names.c
--- ../busybox-1.00+/libbb/u_signal_names.c	2004-03-15 00:28:43.000000000 -0800
+++ ./libbb/u_signal_names.c	2009-06-07 15:08:50.000000000 -0700
@@ -21,6 +21,7 @@
  * USA
  */
 
+#include <sys/types.h>
 #include <signal.h>
 #include <ctype.h>
 #include <string.h>
diff -Bubr ../busybox-1.00+/libbb/vdprintf.c ./libbb/vdprintf.c
--- ../busybox-1.00+/libbb/vdprintf.c	2004-03-15 00:28:43.000000000 -0800
+++ ./libbb/vdprintf.c	2009-06-07 15:08:50.000000000 -0700
@@ -26,7 +26,17 @@
 
 
 #if (__GLIBC__ < 2)
-extern int vdprintf(int d, const char *format, va_list ap)
+int dprintf(int d, const char *fmt, ...) {
+    int ret;
+    va_list ap;
+    
+    va_start (ap, fmt);
+    ret = vdprintf (d, fmt, ap);
+    va_end (ap);
+    return ret;
+}
+
+int vdprintf(int d, const char *format, va_list ap)
 {
 	char buf[BUF_SIZE];
 	int len;
diff -Bubr ../busybox-1.00+/libbb/xconnect.c ./libbb/xconnect.c
--- ../busybox-1.00+/libbb/xconnect.c	2004-04-14 10:51:16.000000000 -0700
+++ ./libbb/xconnect.c	2009-06-07 15:08:50.000000000 -0700
@@ -62,7 +62,7 @@
 int xconnect(struct sockaddr_in *s_addr)
 {
 	int s = socket(AF_INET, SOCK_STREAM, 0);
-	if (connect(s, (struct sockaddr_in *)s_addr, sizeof(struct sockaddr_in)) < 0)
+	if (connect(s, (struct sockaddr *)s_addr, sizeof(struct sockaddr_in)) < 0)
 	{
 		bb_perror_msg_and_die("Unable to connect to remote host (%s)",
 				inet_ntoa(s_addr->sin_addr));
diff -Bubr ../busybox-1.00+/libbb/xgetcwd.c ./libbb/xgetcwd.c
--- ../busybox-1.00+/libbb/xgetcwd.c	2003-05-26 07:06:00.000000000 -0700
+++ ./libbb/xgetcwd.c	2009-06-07 15:08:50.000000000 -0700
@@ -11,6 +11,7 @@
 #include <unistd.h>
 #include <limits.h>
 #include <sys/param.h>
+#include <sys/syslimits.h>
 #include "libbb.h"
 
 /* Amount to increase buffer size by in each try. */
diff -Bubr ../busybox-1.00+/networking/httpd.c ./networking/httpd.c
--- ../busybox-1.00+/networking/httpd.c	2004-10-08 01:03:29.000000000 -0700
+++ ./networking/httpd.c	2009-06-07 15:11:28.000000000 -0700
@@ -1789,11 +1789,15 @@
   FD_ZERO (&s_fd) ;
   FD_SET (a_c_w, &s_fd) ;
 
+#ifdef NESTEDVM
+  //while(read(a_c_w, buf, sizeof(config->buf) > 0));
+#else
   do {
     tv.tv_sec = 2 ;
     tv.tv_usec = 0 ;
     retval = select (a_c_w + 1, &s_fd, NULL, NULL, &tv);
   } while (retval > 0 && (read (a_c_w, buf, sizeof (config->buf)) > 0));
+#endif
 
   shutdown(a_c_r, SHUT_RD);
   close(config->accepted_socket);
@@ -1818,18 +1822,22 @@
 #ifndef CONFIG_FEATURE_HTTPD_USAGE_FROM_INETD_ONLY
 static int miniHttpd(int server)
 {
+#ifndef NESTEDVM
   fd_set readfd, portfd;
 
   FD_ZERO(&portfd);
   FD_SET(server, &portfd);
+#endif
 
   /* copy the ports we are watching to the readfd set */
   while (1) {
+#ifndef NESTEDVM
     readfd = portfd;
 
     /* Now wait INDEFINITELY on the set of sockets! */
     if (select(server + 1, &readfd, 0, 0, 0) > 0) {
       if (FD_ISSET(server, &readfd)) {
+#endif
 	int on;
 	struct sockaddr_in fromAddr;
 
@@ -1838,7 +1846,7 @@
 		       (struct sockaddr *)&fromAddr, &fromAddrLen);
 
 	if (s < 0) {
-	    continue;
+	    break;
 	}
 	config->accepted_socket = s;
 	config->rmt_ip = ntohl(fromAddr.sin_addr.s_addr);
@@ -1872,8 +1880,10 @@
 		exit(0);
 	}
 	close(s);
+#ifndef NESTEDVM
       }
     }
+#endif
   } // while (1)
   return 0;
 }
Only in ./networking: httpd.c.orig
diff -Bubr ../busybox-1.00+/networking/tftp.c ./networking/tftp.c
--- ../busybox-1.00+/networking/tftp.c	2004-09-14 10:24:58.000000000 -0700
+++ ./networking/tftp.c	2009-06-07 15:09:40.000000000 -0700
@@ -320,7 +320,11 @@
 			FD_ZERO(&rfds);
 			FD_SET(socketfd, &rfds);
 
+#ifdef NESTEDVM
+			switch(1) {
+#else
 			switch (select(FD_SETSIZE, &rfds, NULL, NULL, &tv)) {
+#endif
 			case 1:
 				len = recvfrom(socketfd, buf, tftp_bufsize, 0,
 						(struct sockaddr *) &from, &fromlen);
diff -Bubr ../busybox-1.00+/procps/ps.c ./procps/ps.c
--- ../busybox-1.00+/procps/ps.c	2004-03-15 00:29:03.000000000 -0800
+++ ./procps/ps.c	2009-06-07 15:08:54.000000000 -0700
@@ -27,8 +27,10 @@
 #include <fcntl.h>
 #include <ctype.h>
 #include <string.h>
+#ifndef NESTEDVM
 #include <termios.h>
 #include <sys/ioctl.h>
+#endif
 #include "busybox.h"
 #ifdef CONFIG_SELINUX
 #include <fs_secure.h>
diff -Bubr ../busybox-1.00+/Rules.mak ./Rules.mak
--- ../busybox-1.00+/Rules.mak	2004-10-08 03:52:33.000000000 -0700
+++ ./Rules.mak	2009-06-07 15:08:50.000000000 -0700
@@ -158,8 +158,8 @@
     STRIPCMD:=/bin/true -Not_stripping_since_we_are_debugging
 else
     CFLAGS+=$(WARNINGS) $(OPTIMIZATIONS) -D_GNU_SOURCE -DNDEBUG
-    LDFLAGS += -s -Wl,-warn-common
-    STRIPCMD:=$(STRIP) --remove-section=.note --remove-section=.comment
+    LDFLAGS += -Wl,-warn-common
+    STRIPCMD:= true
 endif
 ifeq ($(strip $(CONFIG_STATIC)),y)
     LDFLAGS += --static
Only in .: Rules.mak.orig
diff -Bubr ../busybox-1.00+/shell/ash.c ./shell/ash.c
--- ../busybox-1.00+/shell/ash.c	2004-10-08 02:43:34.000000000 -0700
+++ ./shell/ash.c	2009-06-07 15:08:54.000000000 -0700
@@ -67,7 +67,9 @@
 
 #include <sys/types.h>
 #include <sys/cdefs.h>
+#ifndef NESTEDVM
 #include <sys/ioctl.h>
+#endif
 #include <sys/param.h>
 #include <sys/resource.h>
 #include <sys/stat.h>
@@ -90,8 +92,10 @@
 #include <paths.h>
 #include <setjmp.h>
 #include <signal.h>
+#ifndef NESTEDVM
 #include <stdint.h>
 #include <sysexits.h>
+#endif
 #include <time.h>
 #include <fnmatch.h>
 
@@ -1403,7 +1407,9 @@
 	{ BUILTIN_SPEC_REG      "trap", trapcmd },
 	{ BUILTIN_REGULAR       "true", truecmd },
 	{ BUILTIN_NOSPEC        "type", typecmd },
+#ifndef NESTEDVM
 	{ BUILTIN_NOSPEC        "ulimit", ulimitcmd },
+#endif
 	{ BUILTIN_REGULAR       "umask", umaskcmd },
 #ifdef CONFIG_ASH_ALIAS
 	{ BUILTIN_REGULAR       "unalias", unaliascmd },
@@ -2563,7 +2569,9 @@
 	int i;
 
 	intpending = 0;
+#ifndef NESTEDVM
 	sigsetmask(0);
+#endif
 	i = EXSIG;
 	if (gotsig[SIGINT - 1] && !trap[SIGINT]) {
 		if (!(rootshell && iflag)) {
@@ -6716,7 +6724,11 @@
 #endif
 		}
 		st &= 0x7f;
+#ifdef NESTEDVM
+		col = fmtstr(s, 32, "signal %d",st);
+#else
 		col = fmtstr(s, 32, strsignal(st));
+#endif
 		if (WCOREDUMP(status)) {
 			col += fmtstr(s + col, 16, " (core dumped)");
 		}
@@ -7323,7 +7335,7 @@
 #endif
 	if (block == 0)
 		flags |= WNOHANG;
-	return wait3(status, flags, (struct rusage *)NULL);
+    return waitpid((pid_t)-1,status,flags);
 }
 
 /*
@@ -12717,6 +12729,8 @@
 	return 0;
 }
 
+#ifndef NESTEDVM
+
 /*
  * ulimit builtin
  *
@@ -12898,7 +12912,7 @@
 	}
 	return 0;
 }
-
+#endif /* NESTEDVM */
 
 #ifdef CONFIG_ASH_MATH_SUPPORT
 
Only in ./shell: ash.c.orig
diff -Bubr ../busybox-1.00+/util-linux/more.c ./util-linux/more.c
--- ../busybox-1.00+/util-linux/more.c	2004-03-27 02:02:48.000000000 -0800
+++ ./util-linux/more.c	2009-06-07 15:08:54.000000000 -0700
@@ -32,7 +32,9 @@
 #include <signal.h>
 #include <stdlib.h>
 #include <unistd.h>
+#ifndef NESTEDVM
 #include <sys/ioctl.h>
+#endif
 #include "busybox.h"
 
 
