*** newlib/libc/include/sys/unistd.h~	2011-08-19 09:29:34.000000000 -0500
--- newlib/libc/include/sys/unistd.h	2014-10-14 16:53:02.642073366 -0500
***************
*** 180,188 ****
  #if defined(__CYGWIN__) || defined(__rtems__)
  int     _EXFUN(ttyname_r, (int, char *, size_t)); 
  #endif
  int     _EXFUN(unlink, (const char *__path ));
- int 	_EXFUN(usleep, (useconds_t __useconds));
  int     _EXFUN(vhangup, (void ));
  _READ_WRITE_RETURN_TYPE _EXFUN(write, (int __fd, const void *__buf, size_t __nbyte ));
  
  #ifdef __CYGWIN__
--- 180,187 ----
***************
*** 220,229 ****
  #endif
  
  #if defined(__CYGWIN__) || defined(__rtems__) || defined(__sh__) || defined(__SPU__)
  #if !defined(__INSIDE_CYGWIN__)
- int     _EXFUN(ftruncate, (int __fd, off_t __length));
- int     _EXFUN(truncate, (const char *, off_t __length));
  #endif
  #endif
  
  #if defined(__CYGWIN__) || defined(__rtems__)
--- 219,226 ----
***************
*** 233,245 ****
  #if !(defined  (_WINSOCK_H) || defined (__USE_W32_SOCKETS))
  /* winsock[2].h defines as __stdcall, and with int as 2nd arg */
   int	_EXFUN(gethostname, (char *__name, size_t __len));
  #endif
- char *	_EXFUN(mktemp, (char *));
  #endif
  
  #if defined(__CYGWIN__) || defined(__SPU__) || defined(__rtems__)
- void    _EXFUN(sync, (void));
  #endif
  
  ssize_t _EXFUN(readlink, (const char *__path, char *__buf, size_t __buflen));
  #if defined(__CYGWIN__)
--- 230,240 ----
***************
*** 250,257 ****
--- 245,259 ----
  int	_EXFUN(symlinkat, (const char *, int, const char *));
  int	_EXFUN(unlinkat, (int, const char *, int));
  #endif
  
+ unsigned _EXFUN(usleep, (unsigned int __useconds));
+ int     _EXFUN(ftruncate, (int __fd, off_t __length));
+ int     _EXFUN(truncate, (const char *, off_t __length));
+ char *	_EXFUN(mktemp, (char *));
+ int     _EXFUN(sync, (void));
+ 
+ 
  #define	F_OK	0
  #define	R_OK	4
  #define	W_OK	2
  #define	X_OK	1
