*** newlib/libc/machine/mips/Makefile.in~	2011-12-19 16:03:28.000000000 -0600
--- newlib/libc/machine/mips/Makefile.in	2014-10-14 16:45:52.591940862 -0500
***************
*** 52,61 ****
  ARFLAGS = cru
  lib_a_AR = $(AR) $(ARFLAGS)
  lib_a_LIBADD =
  am_lib_a_OBJECTS = lib_a-setjmp.$(OBJEXT) lib_a-strlen.$(OBJEXT) \
! 	lib_a-strcmp.$(OBJEXT) lib_a-strncpy.$(OBJEXT) \
! 	lib_a-memset.$(OBJEXT) lib_a-memcpy.$(OBJEXT)
  lib_a_OBJECTS = $(am_lib_a_OBJECTS)
  DEFAULT_INCLUDES = -I.@am__isrc@
  depcomp =
  am__depfiles_maybe =
--- 52,60 ----
  ARFLAGS = cru
  lib_a_AR = $(AR) $(ARFLAGS)
  lib_a_LIBADD =
  am_lib_a_OBJECTS = lib_a-setjmp.$(OBJEXT) lib_a-strlen.$(OBJEXT) \
! 	lib_a-strcmp.$(OBJEXT) lib_a-strncpy.$(OBJEXT) 
  lib_a_OBJECTS = $(am_lib_a_OBJECTS)
  DEFAULT_INCLUDES = -I.@am__isrc@
  depcomp =
  am__depfiles_maybe =
*** newlib/libc/string/Makefile.in~	2011-03-28 09:51:28.000000000 -0500
--- newlib/libc/string/Makefile.in	2014-10-14 16:45:18.851773554 -0500
***************
*** 56,65 ****
  lib_a_AR = $(AR) $(ARFLAGS)
  lib_a_LIBADD =
  am__objects_1 = lib_a-bcopy.$(OBJEXT) lib_a-bzero.$(OBJEXT) \
  	lib_a-index.$(OBJEXT) lib_a-memchr.$(OBJEXT) \
! 	lib_a-memcmp.$(OBJEXT) lib_a-memcpy.$(OBJEXT) \
! 	lib_a-memmove.$(OBJEXT) lib_a-memset.$(OBJEXT) \
  	lib_a-rindex.$(OBJEXT) lib_a-strcasecmp.$(OBJEXT) \
  	lib_a-strcat.$(OBJEXT) lib_a-strchr.$(OBJEXT) \
  	lib_a-strcmp.$(OBJEXT) lib_a-strcoll.$(OBJEXT) \
  	lib_a-strcpy.$(OBJEXT) lib_a-strcspn.$(OBJEXT) \
--- 56,65 ----
  lib_a_AR = $(AR) $(ARFLAGS)
  lib_a_LIBADD =
  am__objects_1 = lib_a-bcopy.$(OBJEXT) lib_a-bzero.$(OBJEXT) \
  	lib_a-index.$(OBJEXT) lib_a-memchr.$(OBJEXT) \
! 	lib_a-memcmp.$(OBJEXT) \
! 	lib_a-memmove.$(OBJEXT) \
  	lib_a-rindex.$(OBJEXT) lib_a-strcasecmp.$(OBJEXT) \
  	lib_a-strcat.$(OBJEXT) lib_a-strchr.$(OBJEXT) \
  	lib_a-strcmp.$(OBJEXT) lib_a-strcoll.$(OBJEXT) \
  	lib_a-strcpy.$(OBJEXT) lib_a-strcspn.$(OBJEXT) \
