*** missing~	2009-08-22 06:48:33.000000000 -0500
--- missing	2014-10-22 15:19:20.713544905 -0500
***************
*** 313,321 ****
        test -z "$file" && file=`echo "$infile" | sed 's,.*/,,;s,.[^.]*$,,'`.info
      fi
      # If the file does not exist, the user really needs makeinfo;
      # let's fail without touching anything.
!     test -f $file || exit 1
      touch $file
      ;;
  
    tar*)
--- 313,321 ----
        test -z "$file" && file=`echo "$infile" | sed 's,.*/,,;s,.[^.]*$,,'`.info
      fi
      # If the file does not exist, the user really needs makeinfo;
      # let's fail without touching anything.
! #    test -f $file || exit 1
      touch $file
      ;;
  
    tar*)
*** newlib/libc/include/sys/unistd.h~	2014-10-22 18:02:53.000000000 -0500
--- newlib/libc/include/sys/unistd.h	2014-10-22 18:09:44.011333923 -0500
***************
*** 235,243 ****
  
  #if defined(__CYGWIN__) || defined(__SPU__) || defined(__rtems__)
  #endif
  
! ssize_t _EXFUN(readlink, (const char *__path, char *__buf, size_t __buflen));
  #if defined(__CYGWIN__)
  ssize_t	_EXFUN(readlinkat, (int __dirfd1, const char *__path, char *__buf, size_t __buflen));
  #endif
  int     _EXFUN(symlink, (const char *__name1, const char *__name2));
--- 235,243 ----
  
  #if defined(__CYGWIN__) || defined(__SPU__) || defined(__rtems__)
  #endif
  
! ssize_t _EXFUN(readlink, (const char *__path, char *__buf, int __buflen));
  #if defined(__CYGWIN__)
  ssize_t	_EXFUN(readlinkat, (int __dirfd1, const char *__path, char *__buf, size_t __buflen));
  #endif
  int     _EXFUN(symlink, (const char *__name1, const char *__name2));
